test/3_withdrawals.ts

```diff
@@ -454,6 +464,31 @@ describe('Cashier withdrawals ', () => {
           }
         );
 
+        if (i == 0) {
```

Comment:
sorry, I have a difficulty to follow. What does this "i" stand for?

Reply:
This will probably be easier if we get on a call to explain what is happening in this file. But to put it shortly:

Voucher goes through different states - COMMIT, REDEEM, COMPAIN etc...
You can call technically call withdraw no matter in which state it is, but the outcome is not always the same.
There are three possible payouts:
- if withdraw is called before redeem/refund/exipre nothing is withdraw
- if it is called bewteen redeem/refund/exipre and finalize, only payment is withdrawn
- if it is called after finalize there are two options
  - if withdraw was already called before, only deposits are withdrawn
  - if withdraw was not called before, both deposits and payment are withdrawn
  
Now to test all this possible scenarios, I came up with that procedure, where you put in the states through voucher will go and it then just checks that withdraw behaves as expected in each possible scenario that could happen.

So `i` here represents the index of the possible scenarion. As per design `i==0` is always the scenario, where everybody waits until very end to call the withdraw, resulting in both payments and deposits being withdraw at same time. So in this case events emitted and state changes are a bit different then in any other scenarios and are thus in this conditional.